src/main/kotlin/racingcar/domain/RacingCarGame.kt

```diff
@@ -0,0 +1,31 @@
+package racingcar.domain
+
+import racingcar.util.NumGenerator
+
+private const val RANDOM_NUM_BOUND = 10
+
+class RacingCarGame(
+    val racingCars: List<RacingCar>
+) {
+
+    companion object {
+        fun init(racingCars: List<RacingCar>): RacingCarGame {
+            return RacingCarGame(racingCars)
+        }
+    }
+
+    fun race(numGenerator: NumGenerator): RacingCarGame {
+        val newRacingCars = mutableListOf<RacingCar>()
+        for (racingCar in racingCars) {
+            val randomNum = numGenerator.getNextInt(RANDOM_NUM_BOUND)
+            val nextStepRacingCar = racingCar.moveOrStop(randomNum)
+            newRacingCars.add(nextStepRacingCar)
+        }
```

Comment:
아래와 같이 Collection의 확장함수(= map)을 활용해볼 수도 있을 것같아요. 😃
```suggestion
        val newRacingCars = racingCars.map {
            val randomNum = numGenerator.getNextInt(RANDOM_NUM_BOUND)
            it.moveOrStop(randomNum)
        }
```

Reply:
Collection의 `Iterable<T>.map` 확장함수를 이용하니까 코드가 훨씬 간결해지네요!  
오늘도 덕분에 하나 알아갑니다 🙇‍♂️